### PR TITLE
Fix CI for keras-core 0.1.4

### LINKED ISSUE #1200: Fix breakages with keras-core 0.4.1
keras-core 0.4.1 released a few hours ago, and looks like it broke our CI.

https://github.com/keras-team/keras-nlp/actions/runs/5803058494/job/15730471609

We should investigate and fix.

### PR DESCRIPTION
**Pull request by mattdangerw**
Some changes to keras-core base layer needed some changes.

We need to take in both positional arguments for the masked_lm_head layer (not fully sure why yet). And update an argument to avoid a "_mask" postfix, which is a special argument in the eyes of keras-core.

Fixes #1200.

### CODE REVIEW

File: keras_nlp/models/xlnet/xlnet_encoder.py [hunk 1]
```diff
@@ -259,30 +259,26 @@ def __init__(self, hidden_dim, kernel_initializer_range, **kwargs):
             self.kernel_initializer_range
         )
 
-    def build(self, input_shape):
+    def build(self, inputs_shape):
```
Comment (mattdangerw) [inline keras_nlp/models/xlnet/xlnet_encoder.py:262]: we just need to avoid a postfix of `_maxk` here for the new release.

**Review by jbischof — APPROVED** [review]

**Comment by jbischof** [comment]
Pls address failing tests if this is still relevant

**Pull request merged**